### PR TITLE
BL-3931 More info in Report A Problem

### PR DESCRIPTION
**Pull request by gmartin7**
This change is [Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/1299)

### CODE REVIEW

File: src/BloomExe/MiscUI/ProblemReporterDialog.cs [hunk 1]
```diff
@@ -470,10 +469,34 @@ private string GetFullDescriptionContents(bool appendLog)
...
 			return bldr.ToString();
 		}
 
+		private void GetAdditionalEnvironmentInfo(StringBuilder bldr)
+		{
+			if (Book == null)
+				return;
+
+			bldr.AppendLine("=Additional User Environment Information=");
+			var sizeOrient = Book.GetLayout().SizeAndOrientation;
```
Comment (hatton) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: could throw in GetLayout()

File: src/BloomExe/MiscUI/ProblemReporterDialog.cs [hunk 2]
```diff
@@ -470,10 +469,34 @@ private string GetFullDescriptionContents(bool appendLog)
...
+			bldr.AppendLine("Page Size/Orientation: " + sizeOrient);
+
+			var settings = Book.CollectionSettings;
+			if (settings == null)
+				return; // paranoia, shouldn't happen
```
Comment (hatton) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: How about adding a statement that it was null?

File: src/BloomExe/MiscUI/ProblemReporterDialog.cs [hunk 3]
```diff
@@ -470,10 +469,34 @@ private string GetFullDescriptionContents(bool appendLog)
...
 			bldr.AppendLine();
+			GetAdditionalEnvironmentInfo(bldr);
 			GetStandardErrorReportingProperties(bldr, appendLog);
 			return bldr.ToString();
 		}
 
+		private void GetAdditionalEnvironmentInfo(StringBuilder bldr)
+		{
+			if (Book == null)
+				return;
```
Comment (hatton) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: How about adding a statement that it was null?

Comment (hatton) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: I guess the thinking is that if there is no book then we aren't interested in the collection settings? I sorta buy that...

Comment (gmartin7) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: The main problem is that introducing knowledge of WorkspaceView to ProblemReporterDialog might produce a circular reference. WorkspaceView at least has the Collection Name string.
I also spent way too much time trying to make tests that didn't require UI... unfruitful.

File: src/BloomExe/MiscUI/ProblemReporterDialog.cs [hunk 4]
```diff
@@ -470,10 +474,47 @@ private string GetFullDescriptionContents(bool appendLog)
...
 			GetStandardErrorReportingProperties(bldr, appendLog);
 			return bldr.ToString();
 		}
 
+		private void GetAdditionalEnvironmentInfo(StringBuilder bldr)
+		{
+			bldr.AppendLine("=Additional User Environment Information=");
+			if (Book == null)
+			{
+				if (_projectName != string.Empty)
```
Comment (gmartin7) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: I should probably use string.IsNullOrEmpty here.

**Pull request merged**